src/Microsoft.PowerShell.ConsoleHost/host/msh/ManagedEntrance.cs

```diff
@@ -55,54 +55,15 @@ public static int Start(string consoleFilePath, [MarshalAs(UnmanagedType.LPArray
             Thread.CurrentThread.CurrentUICulture = NativeCultureResolver.UICulture;
             Thread.CurrentThread.CurrentCulture = NativeCultureResolver.Culture;
 
-            RunspaceConfigForSingleShell configuration = null;
-            PSConsoleLoadException warning = null;
-            //      PSSnapInException will cause the control to return back to the native code
-            //      and stuff the EXCEPINFO field with the message of the exception.
-            //      The native code will print this out and exit the process.
-            if (string.IsNullOrEmpty(consoleFilePath))
-            {
-#if DEBUG
-// Special switches for debug mode to allow self-hosting on InitialSessionState instead
-// of runspace configuration...
-                    if (args.Length > 0 && !String.IsNullOrEmpty(args[0]) && args[0].Equals("-iss", StringComparison.OrdinalIgnoreCase))
-                    {
-                        ConsoleHost.DefaultInitialSessionState = InitialSessionState.CreateDefault2();
-                        configuration = null;
-                    }
-                    else if (args.Length > 0 && !String.IsNullOrEmpty(args[0]) && args[0].Equals("-isswait", StringComparison.OrdinalIgnoreCase))
```

Comment:
I just ran into a scenario where I need this flag to help debugging: I want to debug the module path resolution code in `ModuleIntrisics.cs` when starting powershell core in Windows PowerShell. I cannot just start/debug powershell core in VSCode, as I need to start it from Windows PowerShell, and `-isswait` is very helpful in this situation.

So I think we probably should keep this hidden flag for debugging purpose.

Reply:
When I needed it, I needed it even earlier than this, so I never saw this code.

A slightly nicer version does something like:

```c#
while (!System.Diagnostics.Debugger.IsAttached) {
    Thread.Sleep(100);
}
System.Diagnostics.Debugger.Break();
```

This way, you don't need to press a key.

Reply:
I'll put in the updated version Jason is suggesting